main.nf

```diff
@@ -191,67 +238,615 @@ process get_software_versions {
     """
     echo $workflow.manifest.version > v_pipeline.txt
     echo $workflow.nextflow.version > v_nextflow.txt
-    fastqc --version > v_fastqc.txt
-    multiqc --version > v_multiqc.txt
     scrape_software_versions.py > software_versions_mqc.yaml
     """
 }
 
+if (params.mzmlPaths) {
+  // Profile 'test' delivers mzmlPaths
+  Channel
+    .from(params.mzmlPaths)
+    .set { mzml_in }
+}
+else if (!params.mzmldef) {
+  Channel
+    .fromPath(params.mzmls)
+    .map { it -> [it, 'NA'] }
+    .set { mzml_in }
+} else {
+  Channel
+    .from(file("${params.mzmldef}").readLines())
+    .map { it -> it.tokenize('\t') }
+    .set { mzml_in }
+}
 
+mzml_in
+  .tap { sets }
+  .map { it -> [file(it[0]), it[1], it[2] ? it[2] : it[1], it[3] ? it[3] : 'NA' ]} // create file, set plate to setname, and fraction to NA if there is none
```

Comment:
Need a bit more verbose code documentation here: Hard to understand why you do it this way..!